syntax = "proto3";
package patch;





/*
ListHost请求
*/
message ListHostRequest {
    /*
    实例ID列表
    */
    repeated string instanceIds = 1 ;
}

/*
ListHost返回
*/
message ListHostResponse {
message Hosts {
    /*
    主机所在的应用系统名称列表（应用系统 --- 组件 --- 集群 --- 主机）
    */
    repeated string businesses = 1 ;
    /*
    实例 ID
    */
    string instanceId = 2 ;
    /*
    主机名
    */
    string hostname = 3 ;
    /*
    主机 IP
    */
    string ip = 4 ;
    /*
    主机环境
    */
    string _environment = 5 ;
    /*
    操作系统类型（如:Windows）
    */
    string osSystem = 6 ;
    /*
    操作系统架构
    */
    string osArchitecture = 7 ;
    /*
    操作系统版本（如:Windows 10）
    */
    string osVersion = 8 ;
    /*
    操作系统内核发行版本
    */
    string osRelease = 9 ;
    /*
    agent 状态
    */
    string _agentStatus = 10 ;
}
        
    
    /*
    主机列表
    */
    repeated Hosts hosts = 1 ;
}

/*
ListHostApi返回
*/
message ListHostResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    patch.ListHostResponse data = 4 ;
}

